osu.Game/Overlays/BeatmapSet/Info.cs

```diff
@@ -147,13 +151,38 @@ public string Text
                     }
 
                     this.FadeIn(transition_duration);
+
                     textFlow.Clear();
-                    textFlow.AddText(value, s => s.Font = s.Font.With(size: 14));
+                    static void format(SpriteText t) => t.Font = t.Font.With(size: 14);
+
+                    switch (type)
+                    {
+                        case MetadataType.Tags:
+                            string[] tags = value.Split(" ");
+                            for (int i = 0; i <= tags.Length - 1; i++)
+                            {
+                                textFlow.AddLink(tags[i], LinkAction.SearchBeatmapSet, tags[i], null, format);
+
+                                if (i != tags.Length - 1)
+                                    textFlow.AddText(" ", format);
+                            }
+                            break;
+
+                        case MetadataType.Source:
+                            textFlow.AddLink(value, LinkAction.SearchBeatmapSet, value, null, format);
+                            break;
+
+                        default:
+                            textFlow.AddText(value, format);
+                            break;
+                    }
```

Comment:
Still think this could be extracted to a common place. Moreover, you could move `MetadataType` somewhere common too, since it's used by both places but placed under `BeatmapDetails`.